plugins/receiver/http/receiver_test.go

```diff
@@ -0,0 +1,141 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package http
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+
+	common "skywalking/network/common/v3"
+	logging "skywalking/network/logging/v3"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/plugin"
+	_ "github.com/apache/skywalking-satellite/internal/satellite/test"
+	receiver "github.com/apache/skywalking-satellite/plugins/receiver/api"
+	server "github.com/apache/skywalking-satellite/plugins/server/api"
+	httpserver "github.com/apache/skywalking-satellite/plugins/server/http"
+	"github.com/apache/skywalking-satellite/protocol/gen-codes/satellite/protocol"
+)
+
+func TestReceiver_http_RegisterHandler(t *testing.T) {
+	Init()
+	r := initReceiver(make(plugin.Config), t)
+	s := initServer(make(plugin.Config), t)
+	r.RegisterHandler(s.GetServer())
+	err := s.Start()
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	time.Sleep(time.Second)
+	defer func() {
+		if err := s.Close(); err != nil {
+			t.Fatalf("cannot close the http sever: %v", err)
+		}
+	}()
+	for i := 0; i < 10; i++ {
+		data := initData(i)
+		dataBytes, err := proto.Marshal(data)
```

Comment:
should add some test cases for invalid data

Reply:
Done